modules/coinzillaBidAdapter.js

```diff
@@ -39,12 +39,28 @@ export const spec = {
       const sizes = parseSizesInput(bidRequest.params.size || bidRequest.sizes)[0];
       const width = sizes.split('x')[0];
       const height = sizes.split('x')[1];
+      let installed_wallet = '';
+      if (typeof window.ethereum !== 'undefined' || (typeof window.web3 !== 'undefined')) {
+        installed_wallet = "ETH";
+      }
+      if ((typeof window.cardano !== 'undefined')) {
+        installed_wallet = "ADA";
+      }
+      if ((typeof window.BinanceChain !== 'undefined')) {
+        installed_wallet = "BNB";
+      }
+      if ((typeof window.solana !== 'undefined')) {
+        installed_wallet = "SOL";
+      }
+      if ((typeof window.tron !== 'undefined')) {
+        installed_wallet = "TRX";
+      }
```

Comment:
Hi, it appears this detection is very similar to https://github.com/prebid/Prebid.js/blob/4b46054f75ddc7116b23a129062bb473bd06dfc0/modules/hypelabBidAdapter.js#L66

Any chance you could move both detection routines into a utility in /libraries/crypto/index.js ?

Reply:
Hello @patmmccann, we'd love to keep them at this moment as it is to see if there's traction with the web3 targeting. 

Our ad codes are already up and running with this targeting on Coinzilla and that's why we'd love to have it available on our Prebid adapter as soon as possible too. 

PS: Thank you @ChrisHuie for hinting the linter errors, we didn't changed any other logic, we just added the wallet detection, that's why we don't have unit testing on Prebid. 

Our detection is already live at https://coinzillatag.com/lib/display.js

Cheers!

Reply:
You must add unit testing, also you must unify the detection over the two adapters into a library